src/main/java/org/apache/commons/net/ftp/parser/OS400FTPEntryParser.java

```diff
@@ -243,7 +243,7 @@
         + "(?:(\\d+)\\s+)?"           // size, empty for members
         + "(?:(\\S+)\\s+(\\S+)\\s+)?" // date stuff, empty for members
         + "(\\*STMF|\\*DIR|\\*FILE|\\*MEM)\\s+"  // *STMF/*DIR/*FILE/*MEM
-        + "(?:(\\S+)\\s*)?";          // file name, missing, when CWD is a *FILE
+        + "(((\\S+)\\s*)+)?";           // file name, missing, when CWD is a *FILE
```

Comment:
I think the RE is more complicated than it needs to be.

Also it does not handle file names which start with a space, if that is possible.

Reply:
Hi @sebbASF 
Could you please elaborate on the complicated parts of RegEX? I did just the small fix and do not make it more complicated than it's already been

For file started with spaces - I can't find any strict documentation which says it's possible (it's not a unix system), but when I tried to upload such file or rename with RNTO command, the heading spaces just were trimmed in the file name, so I don't think it's supported

Reply:
If you have direct access to the OS, you could try creating a name with a leading space, and see how it appears in FTP.
You may need to escape or quote the space. Similarly with FTP, it's possible that the client drops leading spaces.

"(((\\S+)\\s*)+)?"
could be expressed as
"((\\S+\\s*)+)?"
unless you need to refer to the non-space matches separately.

Unless trailing spaces need to be trimmed, it can be further simplified to:
"(.*)?"
because leading spaces have already been removed.